wgpu-hal/src/vulkan/device.rs

```diff
@@ -2435,6 +2519,10 @@ impl crate::Device for super::Device {
         &self,
         desc: &wgt::QuerySetDescriptor<crate::Label>,
     ) -> Result<super::QuerySet, crate::DeviceError> {
+        // Assume each query is 256 bytes.
+        // On an AMD W6800 with driver version 32.0.12030.9, occlusion queries are 256.
+        self.check_for_oom(true, desc.count as u64 * 256)?;
```

Comment:
I'm not 100% sure about this, there doesn't seem to be any indication in the Vulkan API that query sets reside in a host visible heap (even if this is the case on the AMD card I tested).

Reply:
I was actually surprised they are in the host visible heap rather than the device local one (as they are on D3D12).